lib/src/equatable_utils.dart

```diff
@@ -1,13 +1,13 @@
 import 'package:collection/collection.dart';
 
 /// Returns a `hashCode` for [props].
-int mapPropsToHashCode(Iterable props) =>
-    _finish(props?.fold(0, _combine) ?? 0);
+int mapPropsToHashCode(Iterable? props) =>
+    _finish(props == null ? 0 : props.fold(0, _combine));
```

Comment:
wondering if there is a simpler way to write this

Reply:
A different variant is 
```dart
props?.fold(0, _combine) ?? 0
```
Whether it's simpler is a matter of taste. I find yours easier to *read* because I don't have to consider whether the `.fold` call can return a `null` value too.

Reply:
That’s what I originally had but it was forcing me to ask combine as returning a nullable int

Reply:
Type inference on `fold()` doesn't always produce the type you'd hope for. In this case, I think because the surrounding context allows a nullable int (I'm guessing because of the type of `_finish()` and the `??`), it infers `fold<int>?(...)`. That in turn means it expects `_combine` to take a nullable int.

You can fix this by providing `fold()` with an explicit type argument:

```dart
props?.fold<int>(0, _combine) ?? 0
```

Reply:
Thanks, I’ll give that a shot. Is this behavior working as expected or is it something that can be improved?

Reply:
It's working as expected. This has been a funny corner of type inference since before null safety, you just may have never run into it before. We have talked about making the inference smarter here, but inference is hard to tweak without breaking things and the benefit is relatively marginal.